src/Route.js

```diff
@@ -6,11 +6,14 @@ export function Route(props) {
     exact: !props.parent
   })
 
-  return (
-    match &&
-    props.render({
-      match: match,
-      location: location
-    })
-  )
+  if (!match) return
+
+  var child = props.render({
+    match: match,
+    location: location
+  })
+
+  child.props.key = child.props.key || props.path
```

Comment:
@ForsakenHarmony @zaceno Is this infallible? 

Anyone familiar with ReactRouter, what's going on there? 

I like this though.

Reply:
`props` always gets added by `h` at least, if that's what you're asking

Reply:
Nope, I mean does ReactRouter also add a key to the route component?

Reply:
afaik, no, but they have component methods, which don't depend on elements that might be reused

Reply:
@JorgeBucaran is it infallible? ...well... this:

```jsx
<main>
   <Route path="/show/secrets" render={SecretComponent} />
   <AComponent />
   <AnotherComponent />
   <Route path="/show/secrets" render={AnotherSecretComponent} />
   <ThirdComponent />
</main>
```

I e using the route as a switch to turn on or off multiple sibling components. Then the siblings will be keyed with the same key which of course is no good. ...unless the components have their own keys, in which case those keys will be used instead. So:

A - Before: By default `oncreate` won't get called for routed components. You have to manually key them to make it work. Beginners are required to understand the relationship between keys and lifecycle events. Higher threshold.

B - This proposal: You get automatic keys which make sense for the most part. And hence a lower threshold for beginners. But you run the risk of key-collisions and you don't understand why.

Reply:
What about this?

```js
var key= props.key || (props.path + (props.render ? props.render.name + props.render.length : ""))
```

Reply:
Arrow functions have no name. 😞

Reply:
They do if you assign them to a variable, but anonymously used, they don't, just `""`

Reply:
I guess we could use `props.render.toString()`, but that feels asking for trouble.